src/otx/v2/adapters/torch/dataset.py

```diff
@@ -31,20 +31,20 @@ def _initialize(self) -> None:
     def _build_dataset(
         self,
         subset: str,
-        pipeline: list | dict | None = None,
-        config: str | dict | None = None,
+        pipeline: list | None = None,
+        config: dict | None = None,
     ) -> TorchDataset | None:
         """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
 
         Args:
             subset (str): The subset to build the dataset for.
-            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+            pipeline (list | None, optional): The pipeline to use for the dataset.
```

Comment:
Can you explain why `pipeline` cannot be `dict` but only a `list`? I have the feeling that we do need to re-visit this pipeline approach.

Reply:
Sure. The reason the pipeline used to be a dictionary was for more flexibility (for example, to get labeled and unlabeled together in semi-sl), but build_dataset doesn't need to get them both at the same time, it can get them as a list, and subset_dataloader's pipeline gets them as dict[str, list], but build_dataset gets a list. so subset_dataloader just passes it to buil_dataset as list. This meant that `build_dataset` was accepting both `dict | list` but not actually using them properly, which Jaeguk found and I fixed.

Reply:
subset_dataloader(..., pipeline: dict | list | None) -> Pick a list from a dictionary and pass it -> build_dataset(..., pipeline: list | None)

Reply:
Thanks for clariying this. No another ask then. Would it be possible to add an `Example` section here, so that reader/user would easily get a clearer picture when they read the docstring?

Reply:
Sure, I'll add it.

Reply:
Thanks!

Reply:
I added https://github.com/openvinotoolkit/training_extensions/pull/2599/commits/3eaa8f3bb7446279b4136dc871cc2fe38777b96d

Reply:
Thanks, just a typo suggestion